bundle/phases/initialize.go

```diff
@@ -28,6 +29,7 @@ func Initialize() bundle.Mutator {
 			mutator.ExpandWorkspaceRoot(),
 			mutator.DefineDefaultWorkspacePaths(),
 			mutator.SetVariables(),
+			pythonmutator.ApplyPythonMutator(pythonmutator.ApplyPythonMutatorPhaseInit),
```

Comment:
This spot is intentional because the generated code can still refer to bundle variables and have them resolved in the later stages. However, the variable values are not necessarily known (e.g. in case of lookups, or in case of variables referring to other variables).

If you want to have access to the underlying values then this mutator should run after `ResolveResourceReferences` and before `ResolveVariableReferences`.

Either way, the intention should be captured in a comment to avoid the risk of this being unintentionally reordered later on.